### PR TITLE
service/dap: Add buildFlags error checking and tests in launch requests

### PR DESCRIPTION
**Pull request by polinasok**

### CODE REVIEW

File: pkg/proc/test/support.go [hunk 1]
```diff
@@ -81,6 +81,12 @@ const (
 
 // BuildFixture will compile the fixture 'name' using the provided build flags.
 func BuildFixture(name string, flags BuildFlags) Fixture {
+	return BuildFixtureWithFlags(name, flags, "")
+}
+
+// BuildFixtureWithFlags will compile the fixture 'name' using the provided predefined
+// and custom build flags.
+func BuildFixtureWithFlags(name string, flags BuildFlags, andflags string) Fixture {
```
Comment (eliben) [inline pkg/proc/test/support.go]: Maybe call it customFlags to match the comment? Also, the comment should say what the format for the flags is. "Flags" is plural but the type is `string`, so it should be more specific (space separated, etc.)

**Review by eliben — COMMENTED** [review]
Did you need to add the new fixture to test this? I'm not sure any other existing fixture will fail if you pass these `ldflags`.

**Comment by polinasok** [comment]
I used the flags to modify something within the target program that I could test for, so the program panics and fails the test if the condition is not true. No existing fixture has such logic. This is what the failure looks like on the command line:

```
$ go build buildflagtest.go 
$ ./buildflagtest 
panic: global main.Hello not set via build flags

goroutine 1 [running]:
main.main()
	/Users/polina/delve/_fixtures/buildflagtest.go:11 +0xd8
```

And this is what it would look like from the test:
```
--- FAIL: TestLaunchRequestWithBuildFlags (3.79s)
panic: interface conversion: dap.Message is *dap.StoppedEvent, not *dap.TerminatedEvent [recovered]
	panic: interface conversion: dap.Message is *dap.StoppedEvent, not *dap.TerminatedEvent

goroutine 7 [running]:
testing.tRunner.func1.1(0x135bf00, 0xc0001e0540)
	/usr/local/go/src/testing/testing.go:941 +0x3d0
testing.tRunner.func1(0xc000128360)
	/usr/local/go/src/testing/testing.go:944 +0x3f9
panic(0x135bf00, 0xc0001e0540)
	/usr/local/go/src/runtime/panic.go:967 +0x166
github.com/go-delve/delve/service/dap/daptest.(*Client).ExpectTerminatedEvent(0xc000210040, 0xc000128360, 0xc0004299f0)
	/Users/polina/delve/service/dap/daptest/client.go:98 +0x8b
github.com/go-delve/delve/service/dap.runDebugSession(0xc000128360, 0xc000210040, 0xc00022fd80)
	/Users/polina/delve/service/dap/server_test.go:341 +0xbf
github.com/go-delve/delve/service/dap.TestLaunchRequestWithBuildFlags.func1(0xc000210040, 0x13ccb58, 0xd, 0xc00002c640, 0x47, 0xc000026390, 0x2e, 0xc0000247a0, 0x1d)
	/Users/polina/delve/service/dap/server_test.go:381 +0xa6
github.com/go-delve/delve/service/dap.runTest(0xc000128360, 0x13ccb58, 0xd, 0xc000050f60)
	/Users/polina/delve/service/dap/server_test.go:71 +0x389
github.com/go-delve/delve/service/dap.TestLaunchRequestWithBuildFlags(0xc000128360)
	/Users/polina/delve/service/dap/server_test.go:380 +0x63
testing.tRunner(0xc000128360, 0x13de820)
	/usr/local/go/src/testing/testing.go:992 +0xdc
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1043 +0x357
FAIL	github.com/go-delve/delve/service/dap	4.547s
FAIL
Error: Tests failed.
```

File: pkg/proc/test/support.go (hunk 1 above)
Comment (polinasok) [reply pkg/proc/test/support.go]: I ended with a different rename because I rewrote the comment. Please take a look.

**Review by eliben — APPROVED** [review]

**Pull request closed without merging**